ProteoformSuiteInternal/PSM.cs

```diff
@@ -8,41 +8,43 @@ namespace ProteoformSuiteInternal
-        //public int precursor_charge { get; set; }
-        //public double precursor_mass_error { get; set; }
-        //public PsmType psm_type { get; set; }
+        public string base_sequence { get; set; }
+        public string sequence_with_modifications
+        {
+            get {
+                string _base_sequence = base_sequence;
+                foreach (Ptm mod in modifications.OrderByDescending(m => m.position)) _base_sequence = _base_sequence.Insert(mod.position - 1, "[" + mod.modification.id + "]");
+                return _base_sequence;
+            }
+        }
+        public int start_residue { get; set; }
+        public int stop_residue { get; set; }
+        public string spectrum { get; set; }
+        public string protein_description { get; set; }
+        public string protein_accession { get; set; }
+        public double precursor_mz { get; set; }
+        public int precursor_charge { get; set; }
+        public double precursor_mass_error { get; set; }
+        public List<Ptm> modifications = new List<Ptm>();
 
-        //public Psm (string sequence, string filename, int start_residue, int stop_residue, double total_intensity, double precursor_intensity, double morpheus_score, int spectrum,string protein_description, double precursor_mz, int precursor_charge, double precursor_mass_error, PsmType psm_type )
-        //{
-        //    this.sequence = sequence;
-        //    this.filename = filename;
-        //    this.start_residue = start_residue;
-        //    this.stop_residue = stop_residue;
-        //    this.spectrum = spectrum;
-        //    this.total_intensity = total_intensity;
-        //    this.morpheus_score = morpheus_score;
-        //    this.precursor_intensity = precursor_intensity;
-        //    this.protein_description = protein_description;
-        //    this.precursor_mz = precursor_mz;
-        //    this.precursor_charge = precursor_charge;
-        //    this.precursor_mass_error = precursor_mass_error;
-        //    this.psm_type = psm_type;
-        //}
+        public Psm(string sequence, int start_residue, int stop_residue, List<Ptm> modifications,  string spectrum, string protein_accession, string protein_description, double precursor_mz, int precursor_charge, double precursor_mass_error)
+        {
+            this.base_sequence = sequence;
+            this.start_residue = start_residue;
+            this.stop_residue = stop_residue;
+            this.spectrum = spectrum;
+            this.protein_accession = protein_accession;
+            this.protein_description = protein_description;
+            this.precursor_mz = precursor_mz;
+            this.precursor_charge = precursor_charge;
+            this.precursor_mass_error = precursor_mass_error;
+            this.modifications = modifications;
+        }
     }
 
-    //public enum PsmType
-    //{
-    //    BottomUp,
-    //    TopDown
-    //}
+    public enum PsmType
+    {
+        BottomUp,
+        TopDown
```

Comment:
I'm actually thinking it might make more sense to get rid of this and just count BU data read in as a PSM, and if someone reads in TD data it's always considered a proteoform, now that they're matched up with theoreticals through T-TD. User could decide if they want to match up with experimentals or not. Then again, I do like the idea of a Psmtype where you could read in BU,TD, intact data (maybe neucode labeled?) and see where there was evidence for each family/ID.